### PR TITLE
15_7_12 Simplify user input processing

### PR DESCRIPTION
**Pull request by sergeroshe**

### CODE REVIEW

File: 15_mini_project_work/7_hangman_game.py [hunk 1]
```diff
@@ -181,63 +178,48 @@ def get_prompt(guessed_word):
...
+                    tries_remained -= 1
+            else:
+                print(WRONG_GUESS_MESSAGE)
+                tries_remained -= 1
 
-        tries_remained -= 1
-        game_lost = not tries_remained
+        game_won = input_string == hidden_word or word_completion_list == word_char_list
 
+    #  print_game_result extract function
```
Comment (YanumVlad) [inline 15_mini_project_work/7_hangman_game.py:220]: print_game_result extract function

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/7_hangman_game.py [hunk 2]
```diff
@@ -181,63 +178,48 @@ def get_prompt(guessed_word):
...
+                else:
+                    print(REPEAT_ERROR)
+                    tries_remained -= 1
+            else:
+                print(WRONG_GUESS_MESSAGE)
+                tries_remained -= 1
 
-        tries_remained -= 1
-        game_lost = not tries_remained
+        game_won = input_string == hidden_word or word_completion_list == word_char_list
```
Comment (YanumVlad) [inline 15_mini_project_work/7_hangman_game.py:215]: Remove doubled calc

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/7_hangman_game.py [hunk 3]
```diff
@@ -181,79 +176,63 @@ def get_prompt(guessed_word):
...
 
 def hangman_game(guessed_word):
     print(GREETING)
     print(guessed_word)
 
-    guessed_letters = []  # список уже названных букв
+    guessed_letters = []
     guessed_words = []
-
     word_completion = FILLING_CHAR * len(guessed_word)
```
Comment (YanumVlad) [inline 15_mini_project_work/7_hangman_game.py:234]: Extract to separate function

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/7_hangman_game.py [hunk 4]
```diff
@@ -181,79 +176,63 @@ def get_prompt(guessed_word):
...
     word_completion_list = list(word_completion)
-
     word_char_list = list(guessed_word)
     tries_remained = MAX_TRIES_COUNT
```
Comment (YanumVlad) [inline 15_mini_project_work/7_hangman_game.py:237]: Remove unused var

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/7_hangman_game.py (hunk 4 above)
Comment (sergeroshe) [reply 15_mini_project_work/7_hangman_game.py]: Done

File: 15_mini_project_work/7_hangman_game.py [hunk 5]
```diff
@@ -181,88 +176,70 @@ def get_prompt(guessed_word):
...
+    word_char_completion_list = list(word_completion)
 
-    game_run(tries_remained, guessed_word, guessed_letters, word_char_list, word_completion_list)
+    return word_char_completion_list
 
 
 def main():
     guessed_word = get_word()
-    hangman_game(guessed_word)
+    game_run(MAX_TRIES_COUNT, guessed_word)
```
Comment (YanumVlad) [inline 15_mini_project_work/7_hangman_game.py:242]: remove double meaning: hidden_word / guessed_word

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/7_hangman_game.py [hunk 6]
```diff
@@ -167,102 +166,81 @@ def open_guessed_letters(guessed_letter_idx_list, word_char_completion_list, gue
...
 
-    return game_current_stage
-
 
-def get_prompt(guessed_word):
-    enter_guess_prompt = f'{ENTER_GUESS_PROMPT_P_1}{len(guessed_word)}{ENTER_GUESS_PROMPT_P_2}{COLON_SEP}\n'
-    len_error_message = f'{LEN_ERROR_MESSAGE}{len(guessed_word)}{ENTER_GUESS_PROMPT_P_2}{EXCLAMATION_SIGN}'
+def get_prompt(hidden_word):
+    hidden_word_len = len(hidden_word)
+    enter_guess_prompt = f'{ENTER_GUESS_PROMPT_P_1}{hidden_word_len}{ENTER_GUESS_PROMPT_P_2}{COLON_SEP}\n'
```
Comment (YanumVlad) [inline 15_mini_project_work/7_hangman_game.py:174]: Replace with format call

**Review by YanumVlad — CHANGES_REQUESTED** [review]

File: 15_mini_project_work/7_hangman_game.py (hunk 5 above)
Comment (sergeroshe) [reply 15_mini_project_work/7_hangman_game.py]: Removed

**Review by YanumVlad — APPROVED** [review]

**Pull request merged**